TypeScript error in vivliostyle/table.ts

## Error outputs

```
src/ts/vivliostyle/table.ts(458,5): error TS2322: Type 'TableSlot[]' is not assignable to type 'ElementsOffset[]'.
  Type 'TableSlot' is not assignable to type 'ElementsOffset'.
    Property 'calculateOffset' is missing in type 'TableSlot'.
src/ts/vivliostyle/table.ts(468,50): error TS2345: Argument of type 'TableSlot[]' is not assignable to parameter of type 'ElementsOffset[]'.
src/ts/vivliostyle/table.ts(523,14): error TS2395: Individual declarations in merged declaration 'ElementsOffsetOfTableCell' must be all exported or all local.
src/ts/vivliostyle/table.ts(549,7): error TS2395: Individual declarations in merged declaration 'ElementsOffsetOfTableCell' must be all exported or all local.
src/ts/vivliostyle/table.ts(602,3): error TS2377: Constructors for derived classes must contain a 'super' call.
src/ts/vivliostyle/table.ts(717,3): error TS2416: Property 'afterNonElementNode' in type 'EntireTableLayoutStrategy' is not assignable to the same property in base type 'EdgeSkipper'.
  Type '(state: any) => void' is not assignable to type '(state: LayoutIteratorState) => Result<boolean>'.
    Type 'void' is not assignable to type 'Result<boolean>'.
src/ts/vivliostyle/table.ts(722,3): error TS2416: Property 'startInlineElementNode' in type 'EntireTableLayoutStrategy' is not assignable to the same property in base type 'EdgeSkipper'.
  Type '(state: any) => void' is not assignable to type '(state: LayoutIteratorState) => Result<boolean>'.
    Type 'void' is not assignable to type 'Result<boolean>'.
src/ts/vivliostyle/table.ts(727,3): error TS2416: Property 'afterInlineElementNode' in type 'EntireTableLayoutStrategy' is not assignable to the same property in base type 'EdgeSkipper'.
  Type '(state: any) => void' is not assignable to type '(state: LayoutIteratorState) => Result<boolean>'.
    Type 'void' is not assignable to type 'Result<boolean>'.
src/ts/vivliostyle/table.ts(925,5): error TS2322: Type 'Result<{}>' is not assignable to type 'Result<boolean>'.
src/ts/vivliostyle/table.ts(1027,5): error TS2322: Type 'Result<{}>' is not assignable to type 'Result<boolean>'.
src/ts/vivliostyle/table.ts(1142,52): error TS2345: Argument of type 'EntireTableLayoutStrategy' is not assignable to parameter of type 'LayoutIteratorStrategy'.
  Types of property 'startNonElementNode' are incompatible.
    Type '(state: any) => void' is not assignable to type '(state: LayoutIteratorState) => Result<boolean>'.
      Type 'void' is not assignable to type 'Result<boolean>'.
src/ts/vivliostyle/table.ts(1304,5): error TS2322: Type 'Result<{}>' is not assignable to type 'Result<NodeContext>'.
src/ts/vivliostyle/table.ts(1352,52): error TS2345: Argument of type 'TableLayoutStrategy' is not assignable to parameter of type 'LayoutIteratorStrategy'.
  Types of property 'startNonElementNode' are incompatible.
    Type '(state: any) => void' is not assignable to type '(state: LayoutIteratorState) => Result<boolean>'.
      Type 'void' is not assignable to type 'Result<boolean>'.
src/ts/vivliostyle/table.ts(1355,5): error TS2322: Type 'Result<{}>' is not assignable to type 'Result<NodeContext>'.
src/ts/vivliostyle/table.ts(1395,3): error TS2416: Property 'finishBreak' in type 'TableLayoutProcessor' is not assignable to the same property in base type 'LayoutProcessor'.
  Type '(column: any, nodeContext: any, forceRemoveSelf: any, endOfColumn: any) => Result<{}>' is not assignable to type '(column: Column, nodeContext: NodeContext, forceRemoveSelf: boolean, endOfColumn: boolean) => Result<boolean>'.
    Type 'Result<{}>' is not assignable to type 'Result<boolean>'.
```

## Errors depending on other files

* adapt/csscasc.ts #481 
* adapt/cssparse.ts #471
* adapt/cssstyler.ts #473
* adapt/cssvald.ts #472
* adapt/expr.ts #467
* adapt/layout.ts #475
* adapt/net.ts #470
* adapt/task.ts #466
* adapt/taskutiil.ts #468
* adapt/vgen.ts #474
* adapt/vtree.ts #480 
* adapt/xmldoc.ts #469 
* vivliostyle/layoututil.ts #476 
* vivliostyle/repetitiveelements.ts #478 
* vivliostyle/selectors.ts #479

--- a/src/ts/adapt/base.ts
+++ b/src/ts/adapt/base.ts
@@ -610,10 +610,12 @@ export const mapSize = (obj: Object): number => {
 };
 type Event = {
   type: string,
-  target,
-  currentTarget,
-  preventDefault,
-  newPage
+  target?,
+  currentTarget?,
+  preventDefault?,
+  newPage?,
+  anchorElement?,
+  href?
 };
 
 export {Event};

--- a/src/ts/adapt/csscasc.ts
+++ b/src/ts/adapt/csscasc.ts
@@ -2046,7 +2046,7 @@ export const fixed = {
   'none': ''
 };
 
-export const additiveFormat = (entries: Array, num: number) => {
+export const additiveFormat = (entries: any[], num: number) => {
   const max = (entries[0] as number);
   if (num > max || num <= 0 || num != Math.round(num)) {
     return '';
@@ -3378,12 +3378,15 @@ export class DefineParserHandler extends cssparse.SlaveParserHandler {
 
 export class PropSetParserHandler extends
     cssparse.SlaveParserHandler implements cssvalid.PropertyReceiver {
+  order: number;
+
   constructor(
       scope: exprs.LexicalScope, owner: cssparse.DispatchParserHandler,
       public readonly condition: exprs.Val,
       public readonly elementStyle: ElementStyle,
       public readonly validatorSet: cssvalid.ValidatorSet) {
     super(scope, owner, false);
+    this.order = 0;
   }
 
   /**
@@ -3579,7 +3582,7 @@ export const mergeStyle =
  * @template T
  */
 export const convertToPhysical =
-    (src: {[key: string]: CascadeValue}, dest: {[key: string]: T},
+    <T>(src: {[key: string]: CascadeValue}, dest: {[key: string]: T},
      vertical: boolean, rtl: boolean,
      transform: (p1: string, p2: CascadeValue) => T) => {
       const couplingMap = vertical ?

--- a/src/ts/adapt/cssstyler.ts
+++ b/src/ts/adapt/cssstyler.ts
@@ -637,7 +637,7 @@ export class Styler implements AbstractStyler {
       if (nodeOffset >= this.lastOffset) {
         break;
       }
-      let next = node.firstChild;
+      let next: Node = node.firstChild;
       if (next == null) {
         while (true) {
           next = node.nextSibling;
@@ -775,7 +775,7 @@ export class Styler implements AbstractStyler {
     }
     const context = this.context;
     while (true) {
-      let next = this.last.firstChild;
+      let next: Node = this.last.firstChild;
       if (next == null) {
         while (true) {
           if (this.last.nodeType == 1) {

--- a/src/ts/adapt/layout.ts
+++ b/src/ts/adapt/layout.ts
@@ -285,16 +285,16 @@ export interface BreakPosition {
 /**
  * @abstract
  */
-export class AbstractBreakPosition implements BreakPosition {
+export abstract class AbstractBreakPosition implements BreakPosition {
   /**
    * @abstract
    */
-  findAcceptableBreak(column, penalty) {}
+  abstract findAcceptableBreak(column: Column, penalty: number): vtree.NodeContext;
 
   /**
    * @abstract
    */
-  getMinBreakPenalty() {}
+  abstract getMinBreakPenalty(): number;
 
   calculateOffset(column): {current: number, minimum: number} {
     return calculateOffset(
@@ -837,7 +837,7 @@ export class Column extends vtree.Container {
    * Remove all the exclusion floats.
    */
   killFloats(): void {
-    let c = this.element.firstChild;
+    let c: Node = this.element.firstChild;
     while (c) {
       const nc = c.nextSibling;
       if (c.nodeType == 1) {
@@ -1286,7 +1286,7 @@ export class Column extends vtree.Container {
     if (!floatArea) {
       return task.newResult(result);
     }
-    const frame = task.newFrame('layoutSinglePageFloatFragment');
+    const frame = task.newFrame<SinglePageFloatLayoutResult>('layoutSinglePageFloatFragment');
     let failed = false;
     let i = 0;
     frame
@@ -1396,7 +1396,7 @@ export class Column extends vtree.Container {
     const newFloatAreas = [];
     const newFragments = [];
     let failed = false;
-    const frame = task.newFrame('layoutStashedPageFloats');
+    const frame = task.newFrame<boolean>('layoutStashedPageFloats');
     const self = this;
     let i = 0;
     frame
@@ -1855,7 +1855,7 @@ export class Column extends vtree.Container {
 
   postLayoutBlock(
       nodeContext: vtree.NodeContext, checkPoints: vtree.NodeContext[]) {
-    const hooks: PostLayoutBlockHook[] = plugin.getHooksForName(
+    const hooks: plugin.PostLayoutBlockHook[] = plugin.getHooksForName(
         plugin.HOOKS.POST_LAYOUT_BLOCK);
     hooks.forEach((hook) => {
       hook(nodeContext, checkPoints, this);
@@ -2114,8 +2114,8 @@ export class Column extends vtree.Container {
       orphans = 1;
     } else {
       // Get widows/orphans settings from the block element
-      widows = Math.max((block.inheritedProps['widows'] || 2) - 0, 1);
-      orphans = Math.max((block.inheritedProps['orphans'] || 2) - 0, 1);
+      widows = Math.max((block.inheritedProps['widows'] as number || 2) - 0, 1);
+      orphans = Math.max((block.inheritedProps['orphans'] as number || 2) - 0, 1);
     }
 
     // In case of box-decoration-break: clone, width (or height in vertical
@@ -3191,7 +3191,7 @@ export class Column extends vtree.Container {
    */
   redoLayout(): task.Result<vtree.ChunkPosition> {
     const chunkPositions = this.chunkPositions;
-    let last = this.element.lastChild;
+    let last: Node = this.element.lastChild;
     while (last != this.last) {
       const prev = last.previousSibling;
       if (!(this.element === last.parentNode &&

--- a/src/ts/adapt/vgen.ts
+++ b/src/ts/adapt/vgen.ts
@@ -437,7 +437,7 @@ export class ViewFactory extends base.SimpleEventTarget implements
     const position = (computedStyle['position'] as css.Ident);
     const float = (computedStyle['float'] as css.Ident);
     const displayValues = display.getComputedDislayValue(
-        computedStyle['display'] || css.ident.inline, position, float,
+        computedStyle['display'] as css.Ident || css.ident.inline, position, float,
         this.sourceNode === this.xmldoc.root);
     ['display', 'position', 'float'].forEach((name) => {
       if (displayValues[name]) {
@@ -874,8 +874,7 @@ export class ViewFactory extends base.SimpleEventTarget implements
               if (navParent) {
                 // find the content element
                 let href = null;
-                for (let c = navParent.firstChild; c;
-                     c = c.nextSibling) {
+                for (let c: Node = navParent.firstChild; c; c = c.nextSibling) {
                   if (c.nodeType != 1) {
                     continue;
                   }
@@ -1318,7 +1317,7 @@ export class ViewFactory extends base.SimpleEventTarget implements
 
   private findAndProcessRepeatingElements(
       element: Element, styler: cssstyler.AbstractStyler) {
-    for (let child = element.firstChild; child; child = child.nextSibling) {
+    for (let child: Node = element.firstChild; child; child = child.nextSibling) {
       if (child.nodeType !== 1) {
         continue;
       }
@@ -1452,7 +1451,7 @@ export class ViewFactory extends base.SimpleEventTarget implements
   /**
    * @override
    */
-  setCurrent(nodeContext, firstTime, atUnforcedBreak) {
+  setCurrent(nodeContext, firstTime, atUnforcedBreak = false) {
     this.nodeContext = nodeContext;
     if (nodeContext) {
       this.sourceNode = nodeContext.sourceNode;
@@ -1983,19 +1982,19 @@ export class Viewport {
     let outerZoomBox = this.root.firstElementChild;
     if (!outerZoomBox) {
       outerZoomBox = this.document.createElement('div');
-      outerZoomBox.setAttribute('data-vivliostyle-outer-zoom-box', true);
+      outerZoomBox.setAttribute('data-vivliostyle-outer-zoom-box', 'true');
       this.root.appendChild(outerZoomBox);
     }
     let contentContainer = outerZoomBox.firstElementChild;
     if (!contentContainer) {
       contentContainer = this.document.createElement('div');
-      contentContainer.setAttribute('data-vivliostyle-spread-container', true);
+      contentContainer.setAttribute('data-vivliostyle-spread-container', 'true');
       outerZoomBox.appendChild(contentContainer);
     }
     let layoutBox = outerZoomBox.nextElementSibling;
     if (!layoutBox) {
       layoutBox = this.document.createElement('div');
-      layoutBox.setAttribute('data-vivliostyle-layout-box', true);
+      layoutBox.setAttribute('data-vivliostyle-layout-box', 'true');
       this.root.appendChild(layoutBox);
     }
     this.outerZoomBox = (outerZoomBox as HTMLElement);

--- a/src/ts/adapt/vtree.ts
+++ b/src/ts/adapt/vtree.ts
@@ -100,7 +100,7 @@ export const actions = {
 };
 
 export const makeListener =
-    (refs: Element[], action: string): Function|null => {
+    (refs: Element[], action: string): EventListener | null => {
       const actionFn = actions[action];
       if (actionFn) {
         return () => {
@@ -133,7 +133,7 @@ export class Page extends base.SimpleEventTarget {
   position: LayoutPosition = null;
   offset: number = -1;
   side: constants.PageSide|null = null;
-  fetchers: taskutil.Fetcher[] = [];
+  fetchers: taskutil.Fetcher<{}>[] = [];
   marginBoxes: {
     top: {[key: string]: Container},
     bottom: {[key: string]: Container},
@@ -169,7 +169,7 @@ export class Page extends base.SimpleEventTarget {
   setAutoPageWidth(isAuto: boolean) {
     this.isAutoPageWidth = isAuto;
     if (isAuto) {
-      this.container.setAttribute(Page.AUTO_PAGE_WIDTH_ATTRIBUTE, true);
+      this.container.setAttribute(Page.AUTO_PAGE_WIDTH_ATTRIBUTE, 'true');
     } else {
       this.container.removeAttribute(Page.AUTO_PAGE_WIDTH_ATTRIBUTE);
     }
@@ -178,7 +178,7 @@ export class Page extends base.SimpleEventTarget {
   setAutoPageHeight(isAuto: boolean) {
     this.isAutoPageHeight = isAuto;
     if (isAuto) {
-      this.container.setAttribute(Page.AUTO_PAGE_HEIGHT_ATTRIBUTE, true);
+      this.container.setAttribute(Page.AUTO_PAGE_HEIGHT_ATTRIBUTE, 'true');
     } else {
       this.container.removeAttribute(Page.AUTO_PAGE_HEIGHT_ATTRIBUTE);
     }
@@ -1012,6 +1012,10 @@ export class LayoutPosition {
    * flowPositions is built up to this offset.
    */
   highestSeenOffset: number = 0;
+
+  // FIXME: This properties seem to be not used
+  highestSeenNode: any;
+  lookupPositionOffset: any;
 
   clone(): LayoutPosition {
     const newcp = new LayoutPosition();

--- a/src/ts/vivliostyle/layoututil.ts
+++ b/src/ts/vivliostyle/layoututil.ts
@@ -44,37 +44,27 @@ export class LayoutIteratorStrategy {
     };
   }
 
-  startNonDisplayableNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  startNonDisplayableNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  afterNonDisplayableNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  afterNonDisplayableNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  startIgnoredTextNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  startIgnoredTextNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  afterIgnoredTextNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  afterIgnoredTextNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  startNonElementNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  startNonElementNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  afterNonElementNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  afterNonElementNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  startInlineElementNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  startInlineElementNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  afterInlineElementNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  afterInlineElementNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  startNonInlineElementNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  startNonInlineElementNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  afterNonInlineElementNode(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  afterNonInlineElementNode(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  finish(state: LayoutIteratorState): undefined|task.Result<boolean> {}
+  finish(state: LayoutIteratorState): void | task.Result<boolean> {}
 }
 
 export class LayoutIterator {
@@ -169,13 +159,11 @@ export class EdgeSkipper extends LayoutIteratorStrategy {
       super()
   }
 
-  startNonInlineBox(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  startNonInlineBox(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  endEmptyNonInlineBox(state: LayoutIteratorState): undefined
-      |task.Result<boolean> {}
+  endEmptyNonInlineBox(state: LayoutIteratorState): void | task.Result<boolean> {}
 
-  endNonInlineBox(state: LayoutIteratorState): undefined|task.Result<boolean> {}
+  endNonInlineBox(state: LayoutIteratorState): void | task.Result<boolean> {}
 
   initialState(initialNodeContext: vtree.NodeContext): LayoutIteratorState {
     return {
@@ -389,7 +377,7 @@ export class PseudoColumn {
 /**
  * @abstract
  */
-export class AbstractLayoutRetryer {
+export abstract class AbstractLayoutRetryer {
   initialBreakPositions: BreakPosition[] = null;
   initialStateOfFormattingContext: any = null;
   initialPosition: any;
@@ -404,7 +392,7 @@ export class AbstractLayoutRetryer {
   private tryLayout(nodeContext: vtree.NodeContext, column: Column):
       task.Result<vtree.NodeContext> {
     const frame =
-        task.newFrame('vivliostyle.layoututil.AbstractLayoutRetryer.tryLayout');
+        task.newFrame<vtree.NodeContext>('vivliostyle.layoututil.AbstractLayoutRetryer.tryLayout');
     this.saveState(nodeContext, column);
     const mode = this.resolveLayoutMode(nodeContext);
     mode.doLayout(nodeContext, column).then(function(positionAfter) {
@@ -426,7 +414,7 @@ export class AbstractLayoutRetryer {
   /**
    * @abstract
    */
-  resolveLayoutMode(nodeContext: vtree.NodeContext): LayoutMode {}
+  abstract resolveLayoutMode(nodeContext: vtree.NodeContext): LayoutMode;
 
   prepareLayout(nodeContext: vtree.NodeContext, column: Column) {}
 

--- a/src/ts/vivliostyle/layoututil.ts
+++ b/src/ts/vivliostyle/layoututil.ts
@@ -155,7 +155,7 @@ export class LayoutIterator {
 }
 
 export class EdgeSkipper extends LayoutIteratorStrategy {
-  constructor(protected readonly leadingEdge: boolean) {
+  constructor(protected readonly leadingEdge?: boolean) {
       super()
   }
 

--- a/src/ts/vivliostyle/repetitiveelements.ts
+++ b/src/ts/vivliostyle/repetitiveelements.ts
@@ -374,7 +374,7 @@ export class RepetitiveElements implements ElementsOffset {
 /**
  * @abstract
  */
-export class LayoutEntireBlock implements LayoutMode {
+export abstract class LayoutEntireBlock implements LayoutMode {
   formattingContext: any;
 
   constructor(formattingContext: RepetitiveElementsOwnerFormattingContext) {
@@ -384,7 +384,8 @@ export class LayoutEntireBlock implements LayoutMode {
   /**
    * @override
    */
-  doLayout(nodeContext, column) {}
+  abstract doLayout(nodeContext: vtree.NodeContext, column: layout.Column):
+      task.Result<vtree.NodeContext>;
 
   /**
    * @override
@@ -410,7 +411,7 @@ export class LayoutEntireBlock implements LayoutMode {
 /**
  * @abstract
  */
-export class LayoutFragmentedBlock implements LayoutMode {
+export abstract class LayoutFragmentedBlock implements LayoutMode {
   formattingContext: any;
 
   constructor(formattingContext: RepetitiveElementsOwnerFormattingContext) {
@@ -420,7 +421,8 @@ export class LayoutFragmentedBlock implements LayoutMode {
   /**
    * @override
    */
-  doLayout(nodeContext, column) {}
+  abstract doLayout(nodeContext: vtree.NodeContext, column: layout.Column):
+      task.Result<vtree.NodeContext>;
 
   /**
    * @override
@@ -460,7 +462,7 @@ export class LayoutFragmentedOwnerBlock extends
   constructor(
       formattingContext: RepetitiveElementsOwnerFormattingContext,
       public readonly processor: RepetitiveElementsOwnerLayoutProcessor) {
-    LayoutFragmentedBlock.call(this, formattingContext);
+    super(formattingContext);
   }
 
   /**
@@ -609,7 +611,9 @@ export class EntireBlockLayoutStrategy extends
   constructor(
       public readonly formattingContext:
           RepetitiveElementsOwnerFormattingContext,
-      public readonly column: layout.Column) {}
+      public readonly column: layout.Column) {
+    super();
+  }
 
   /**
    * @override
@@ -671,7 +675,9 @@ export class FragmentedBlockLayoutStrategy extends
   constructor(
       public readonly formattingContext:
           RepetitiveElementsOwnerFormattingContext,
-      public readonly column: layout.Column) {}
+      public readonly column: layout.Column) {
+    super();
+  }
 }
 
 export class RepetitiveElementsOwnerLayoutProcessor extends
@@ -725,7 +731,7 @@ export class RepetitiveElementsOwnerLayoutProcessor extends
       task.Result<vtree.NodeContext> {
     const formattingContext = getRepetitiveElementsOwnerFormattingContext(
         nodeContext.formattingContext);
-    const frame = task.newFrame('BlockLayoutProcessor.doInitialLayout');
+    const frame = task.newFrame<vtree.NodeContext>('BlockLayoutProcessor.doInitialLayout');
     this.layoutEntireBlock(nodeContext, column).thenFinish(frame);
     return frame.result();
   }

--- a/src/ts/vivliostyle/selectors.ts
+++ b/src/ts/vivliostyle/selectors.ts
@@ -20,7 +20,7 @@ import * as asserts from './asserts';
 
 import * as base from '../adapt/base';
 
-import {CascadeValue, ElementStyle, matchANPlusB, mergeStyle, getStyleMap} from '../adapt/csscasc';
+import {CascadeValue, ElementStyle, matchANPlusB, mergeStyle, getViewConditionalStyleMap} from '../adapt/csscasc';
 import {Context} from '../adapt/expr';
 import {Column, getElementHeight} from '../adapt/layout';
 import {FragmentLayoutConstraint} from '../adapt/layout';
@@ -103,7 +103,7 @@ export const mergeViewConditionalStyles =
 
 export const forEachViewConditionalStyles =
     (style: ElementStyle, callback: (p1: ElementStyle) => any) => {
-      const viewConditionalStyles = getStyleMap(style, '_viewConditionalStyles');
+      const viewConditionalStyles = getViewConditionalStyleMap(style);
       if (!viewConditionalStyles) {
         return;
       }

--- a/src/ts/vivliostyle/table.ts
+++ b/src/ts/vivliostyle/table.ts
@@ -468,7 +468,7 @@ export class TableFormattingContext extends
           cellFragment.pseudoColumn.getColumn(), repetitiveElements);
       collected.push(cellFragment);
       return repetitiveElements;
-    }, []);
+    }, [] as repetitiveelements.ElementsOffset[]);
   }
 
   collectElementsOffsetOfHighestColumn(): repetitiveelements.ElementsOffset[] {
@@ -546,7 +546,6 @@ export class ElementsOffsetOfTableCell implements
     return maxOffset;
   }
 }
-const ElementsOffsetOfTableCell = ElementsOffsetOfTableCell;
 
 function getTableFormattingContext(formattingContext: vtree.FormattingContext):
     TableFormattingContext {
@@ -601,7 +600,9 @@ export class EntireTableLayoutStrategy extends
 
   constructor(
       public readonly formattingContext: TableFormattingContext,
-      public readonly column: layout.Column) {}
+      public readonly column: layout.Column) {
+        super();
+      }
 
   /**
    * @override
@@ -853,7 +854,7 @@ export class TableLayoutStrategy extends layoututil.EdgeSkipper {
     const layoutContext = this.column.layoutContext;
     const currentRow = state.nodeContext;
     currentRow.viewNode.parentNode.removeChild(currentRow.viewNode);
-    const frame = task.newFrame('layoutRowSpanningCellsFromPreviousFragment');
+    const frame = task.newFrame<boolean>('layoutRowSpanningCellsFromPreviousFragment');
     let cont = task.newResult(true);
     let spanningCellRowIndex = 0;
     const occupiedSlotIndices = [];
@@ -996,7 +997,7 @@ export class TableLayoutStrategy extends layoututil.EdgeSkipper {
     const afterNodeContext = nodeContext.copy().modify();
     afterNodeContext.after = true;
     state.nodeContext = afterNodeContext;
-    const frame = task.newFrame('startTableCell');
+    const frame = task.newFrame<boolean>('startTableCell');
     let cont;
     if (this.hasBrokenCellAtSlot(cell.anchorSlot.columnIndex)) {
       const cellBreakPosition =
@@ -1275,7 +1276,7 @@ export class TableLayoutProcessor implements layout.LayoutProcessor {
     asserts.assert(nodeContext.sourceNode);
     const tableLayoutOption = getTableLayoutOption(nodeContext.sourceNode);
     clearTableLayoutOptionCache(nodeContext.sourceNode);
-    const frame = task.newFrame('TableLayoutProcessor.doInitialLayout');
+    const frame = task.newFrame<vtree.NodeContext>('TableLayoutProcessor.doInitialLayout');
     const initialNodeContext = nodeContext.copy();
     this.layoutEntireTable(nodeContext, column)
         .then(function(nodeContextAfter) {
@@ -1350,7 +1351,7 @@ export class TableLayoutProcessor implements layout.LayoutProcessor {
     this.addColGroups(formattingContext, rootViewNode, firstChild);
     const strategy = new TableLayoutStrategy(formattingContext, column);
     const iterator = new layoututil.LayoutIterator(strategy, column.layoutContext);
-    const frame = task.newFrame('TableFormattingContext.doLayout');
+    const frame = task.newFrame<vtree.NodeContext>('TableFormattingContext.doLayout');
     iterator.iterate(nodeContext).thenFinish(frame);
     return frame.result();
   }
@@ -1408,7 +1409,7 @@ export class TableLayoutProcessor implements layout.LayoutProcessor {
             formattingContext.getRowSpanningCellsOverflowingTheRow(rowIndex);
       }
       if (cells.length) {
-        const frame = task.newFrame('TableLayoutProcessor.finishBreak');
+        const frame = task.newFrame<boolean>('TableLayoutProcessor.finishBreak');
         let i = 0;
         frame
             .loopWithFrame((loopFrame) => {
@@ -1558,7 +1559,6 @@ export class LayoutEntireTable extends
    * @override
    */
   doLayout(nodeContext, column) {
-    super.doLayout(nodeContext, column);
     return this.processor.doInitialLayout(nodeContext, column);
   }
 }

